### ISSUE TITLE
Reduced TAX not calculated on shipping location

### ISSUE THREAD

**Issue by wespiremedia** [woocommerce/woocommerce#20811]
**Describe the bug**
In our shop we've different TAX rules per country. For example, we don't calculate TAX when delivery is done in Switzerland but do calculate 21% when delivery is done in Germany. 

We've configured TAX calculation based on shipping address. 
So, if for example, an order is placed with billing in Switzerland and shipping to Germany, TAX should be applied. 

We've configured the calculation in this way: 
![screenshot 2018-07-16 09 53 16](https://user-images.githubusercontent.com/14851445/42747880-61adc7a0-88de-11e8-9b34-d1733f3e58f9.png)

We're facing the situation that calculation of TAX is OK for standard rate, but products with reduced rate TAX class are still without TAX. The calculation rules for the reduced rate are exactly the same as the screenshot above (but with 21 replaced by 6 ofcourse). 

Screenshot of an order: 
![screenshot 2018-07-16 09 57 33](https://user-images.githubusercontent.com/14851445/42747988-bdf7eefa-88de-11e8-87ca-c37914caf954.png)

**To Reproduce**
1. Create two products, one with standard rate shipping and one with reduced rate shipping.
2. Set TAX calculation to 'Based on shipping address'
3. Create TAX rules as provided in the screenshot above. 
4. Add two products to your cart. Set billing address to Germany and uncheck 'send to different address'. Tax of both rates should be applied to order. 
5. Switch billing address to Switzerland, check 'send to different address' and set billing address to Germany. 
6. As you can see now, standard rate tax is applied but reduced rate tax isn't. 

**Screenshots**
See above

**Expected behavior**
In the situation specified above, both standard as reduced rate should be applied. 

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by kloon** [comment]
Chnage the priority of the rule with the 0% tax rate to be higher. Taxes are matched based on priority, and if the wildcard rule matches first it will not continue to match the more specific rules beneath it.

**Comment by wespiremedia** [comment]
I understand the ordering by priority, but in this case that can't be the problem (but tested it as well). 

Orders to Switzerland are without tax. If you set billing address to Switzerland the VAT is 0%. But if I select a different delivery address in Germany, the VAT is still 0%. Germany isn't specified as 0%, so VAT should be applied by the wildcard rule.

**Comment by kloon** [comment]
@wespiremedia kindly provide me with a screenshot of your reduced rate tax table so I can do some testing on my end to try and replicate this.

**Comment by wespiremedia** [comment]
Here you are: 
![screenshot 2018-07-18 11 21 52](https://user-images.githubusercontent.com/14851445/42872269-c76a5212-8a7c-11e8-8af3-56eadd1f0566.png)

**Comment by kloon** [comment]
OK did some debugging and everything works fine for me based on the information your provided. When I set my billing address to Switzerland I get no taxes http://cld.wthms.co/qz4UYl

Then when I switch to Germany I get both normal and reduced rates as I have a product of each in the cart http://cld.wthms.co/m0mWw8

What I did notice was that you have your reduced tax names set to the same as the standard rates, it will then combine the rates into one line, but the totals are still the same.

Check if you have any Javascript errors on your checkout page which could prohibit the totals from updating when you change the country, also try testing with all plugins except WooCommerce disabled and test using the Twentyseventeen and Storefront themes.

I know you checked the error occurs with Storefront but you left out the system status report so hard to tell what is happening on your setting.

I have my taxes setup just like you have in the screenshots and have the tax settings defined as you explained and it shows the correct taxes for me as per the screenshots above.
